src/prefect/infrastructure/kubernetes.py

```diff
@@ -587,27 +588,58 @@ def _watch_job(self, job_name: str) -> int:
                 )
-                for log in logs.stream():
-                    print(log.decode().rstrip())
+                try:
+                    for log in logs.stream():
+                        print(log.decode().rstrip())
+                except Exception:
+                    self.logger.warning(
+                        "Error occurred while streaming logs - "
+                        "Job will continue to run but logs will "
+                        "no longer be streamed to stdout.",
+                        exc_info=True,
+                    )
 
-        # Wait for job to complete
-        # if `job_watch_timeout_seconds` is None, no timeout will be enforced
         self.logger.debug(f"Job {job_name!r}: Starting watch for job completion")
-        watch = kubernetes.watch.Watch()
-        with self.get_batch_client() as batch_client:
-            for event in watch.stream(
-                func=batch_client.list_namespaced_job,
-                field_selector=f"metadata.name={job_name}",
-                namespace=self.namespace,
-                timeout_seconds=self.job_watch_timeout_seconds,
+        start_time = pendulum.now("utc")
+        completed = False
+        while not completed:
+            elapsed = (pendulum.now("utc") - start_time).in_seconds()
+            if (
+                self.job_watch_timeout_seconds is not None
+                and elapsed > self.job_watch_timeout_seconds
             ):
-                if event["object"].status.completion_time:
-                    watch.stop()
-                    break
-            else:
-                self.logger.error(f"Job {job_name!r}: Job did not complete.")
+                self.logger.error(
+                    f"Job {job_name!r}: Job timed out " f"after {elapsed}s."
+                )
                 return -1
 
+            watch = kubernetes.watch.Watch()
+            with self.get_batch_client() as batch_client:
+                remaining_timeout = (
+                    (  # subtract previous watch time
+                        self.job_watch_timeout_seconds - elapsed
+                    )
+                    if self.job_watch_timeout_seconds
+                    else None
+                )
+
+                for event in watch.stream(
+                    func=batch_client.list_namespaced_job,
+                    field_selector=f"metadata.name={job_name}",
+                    namespace=self.namespace,
+                    timeout_seconds=remaining_timeout,
+                ):
+                    if event["object"].status.completion_time:
+                        if not event["object"].status.succeeded:
+                            self.logger.error(f"Job {job_name!r}: Job failed.")
+                            return -1
+                        completed = True
+                        watch.stop()
+                        break
+                else:
+                    self.logger.error(f"Job {job_name!r}: Job did not complete.")
```

Comment:
Let's include the timeout here 
```suggestion
                    self.logger.error(f"Job {job_name!r}: Job did not complete within timeout of {self.job_watch_timeout_seconds}s.")
```